### PR TITLE
add(katana): monitoring/profiling - Prometheus monitoring endpoints 

### PR DESCRIPTION
**Pull request by hel-kame**
Implementation of `[Metrics]` derive macro, database integration for metrics, example of metrics integration using `inserted_transactions` in `TransactionPool` function.

What I need to do now is initialize the database that will store all the metrics, or go for another solution. In both cases, the aim is to make their implementation drastically easier.

### CODE REVIEW

File: crates/katana/core/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//! Transaction pool metrics.
```
Comment (glihm) [inline crates/katana/core/src/metrics.rs:1]: We want this module to be the `Core metrics` more than the `Transaction pool metrics`.
In general from what `reth` is doing is having a metrics module per crate, which seems to be appropriate in our case too.

**Review by glihm — COMMENTED** [review]
Hey @hel-kame thanks for the contribution here!
Even if it's still a draft, some comments on the on-going work:

1. It could be great to have the documentation related to the code coming from `reth`. You may also guide us through the design to ensure that a new dev can easily contribute to the common metrics effort. This [README](https://github.com/paradigmxyz/reth/blob/ae329773bceac769d845bea7196043e19b18a394/docs/design/metrics.md) and this [README](https://github.com/paradigmxyz/reth/blob/ae329773bceac769d845bea7196043e19b18a394/book/run/observability.md#L3) can be a good start, adjusting it to dojo.
2. I don't know which text editor you're using, a bunch of files have missing new lines.

Please re-request the review once more changes are pushed to dive again in the PR with a functional endpoint to test different metrics.

Looking forward the coming work on that, the observability of the dojo stack will be very great with the metrics being easily managed.

**Comment by kariy** [comment]
> I had to rebase my main branch to sync the latest persistence commits, I'm reopening a PR on the branch dedicated to my work.

Hi, when are you planning on reopening the PR? I would like to help working on this. I'm thinking of getting metrics set up for `katana` as soon as possible as it would be helpful for development.

**Comment by hel-kame** [comment]
As planned and announced, this week and in the days to come. First, a v1 without persistence. Shall we talk about it?

**Comment by kariy** [comment]
> As planned and announced, this week and in the days to come. First, a v1 without persistence. 

I'm fine without persistence. If you can open a draft PR would be nice to follow along.

> Shall we talk about it?

Sure. In this issue thread or somewhere else? Discord ?

**Comment by broody** [comment]
Hey guys, any updates on the new draft PR? Would also love to follow along as it's an important feature for slot's backend services.

**Comment by hel-kame** [comment]
> Hey guys, any updates on the new draft PR? Would also love to follow along as it's an important feature for slot's backend services.

hey, i've had a few personal problems but everything's fine. I'm reopening the new PR draft tomorrow (friday).

**Pull request closed without merging**